my-app/src/App.tsx

```diff
@@ -4,20 +4,24 @@ import { Title } from "./Components/Title";
 import { User } from "./Components/User";
 import { PostsList } from "./Components/PostsList";
 import { Navigation } from "./Components/Navigation";
+import { ChangeThemeContext, ThemeContext } from "./Components/Context/themeContext";
+import { useState } from "react";
 
-export const App = () => (
+export const App = () => {
 
-  <div>
-    {/* <Title>Sign In</Title> */}
-    {/* <User username="Artem Malkin"></User> */}
-    {/* <ButtonHamburger handleClick={() => console.log('Ok')} collapsed></ButtonHamburger> */}
-    {/* <Tabs tabsArr={['All', 'My favorites', 'Popular']} activeTab="All" disabledTabs={['Popular']} /> */}
-    {/* <Tabs tabsArr={['All', 'My favorites', 'Popular']} activeTab="My favorites" /> */}
-    {/* <PostsList /> */}
+  const [theme, setTheme] = useState('light')
+  const themeToSet = theme === 'light' ? 'dark' : 'light';
 
-    <Navigation />
-  </div>
-);
+  console.log(theme)
+
+  return (
+    <ThemeContext.Provider value={theme}>
+      <ChangeThemeContext.Provider value={() => { setTheme(themeToSet) }}>
```

Comment:
try to move Providers and corresponding state to separate components, it can reduce App rerender
You can compare results in ReactDevTools